Tests/accounts-journey/unit-tests/AccountsJourneyObservabilityTests.swift

```diff
@@ -0,0 +1,131 @@
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        userActionSpy = nil
+        screenViewEventsSpy = nil
+    }
+    
+    //MARK: -  AccountsList Events
+    
+    func test_noUserActionEventIsPublished_whenAccountsListViewModel_is_initialized() {
+        userActionSpy = TrackerSpy<UserActionEvent>()
+        expect(userActionSpy!, toReceive: [], when: {
+            _ = AccountsListViewModel()
+        })
+    }
+    
+    func test_noScreenViewEventIsPublished_whenAccountListViewModel_is_initialized() {
+        screenViewEventsSpy = TrackerSpy<ScreenViewEvent>()
+        expect(screenViewEventsSpy!, toReceive: [], when: {
+            _ = AccountsListViewModel()
+        })
+    }
+    
+    func test_accountsScreenEventIsPublished_whenTheAccountsViewLoads() {
+        screenViewEventsSpy = TrackerSpy<ScreenViewEvent>()
+        let event = AccountsJourney.Tracker.event(forScreen: .accounts)
+        
+        expect(screenViewEventsSpy!, toReceive: [event], when: {
+            AccountsListViewModel().onEvent(.getAccounts)
+        })
+    }
+    
+    func test_refresh_accountsUserActionEvent_isPublished_whenTheRefreshEventIsEmitted() {
+        userActionSpy = TrackerSpy<UserActionEvent>()
+        let event = AccountsJourney.Tracker.event(forUserAction: .refresh_accounts)
+        
+        expect(userActionSpy!, toReceive: [event], when: {
+            AccountsListViewModel().onEvent(.refresh)
+        })
+    }
+    
+    func test_search_accountsUserActionEvent_isPublished_whenTheSearchEventIsEmitted() {
+        userActionSpy = TrackerSpy<UserActionEvent>()
+        let event = AccountsJourney.Tracker.event(forUserAction: .search_accounts)
+        
+        expect(userActionSpy!, toReceive: [event], when: {
+            AccountsListViewModel().onEvent(.search(""))
+        })
+    }
+    
+    
+    // MARK: - AccountDetails Tests
+    func test_noUserActionEventIsPublished_whenAccountDetailsViewModel_is_initialized() {
+        userActionSpy = TrackerSpy<UserActionEvent>()
+        expect(userActionSpy!, toReceive: [], when: {
+            _ = AccountDetailsViewModel()
+        })
+    }
+    
+    func test_noScreenViewEventIsPublished_whenAccountDetailsViewModel_is_initialized() {
+        screenViewEventsSpy = TrackerSpy<ScreenViewEvent>()
+        expect(screenViewEventsSpy!, toReceive: [], when: {
+            _ = AccountDetailsViewModel()
+        })
+    }
+    
+    func test_accountsDetailsScreenEventIsPublished_whenTheAccountsDetailsViewLoads() {
+        screenViewEventsSpy = TrackerSpy<ScreenViewEvent>()
+        let event = AccountsJourney.Tracker.event(forScreen: .account_details)
+        expect(screenViewEventsSpy!, toReceive: [event], when: {
+            AccountDetailsViewModel().onEvent(.getAccountDetails(""))
+        })
+    }
+    
+    // MARK: - Private
+    
+    private class TrackerSpy<S>: NSObject, Tracker {
```

Comment:
Hmm, I like the concept of these two methods, think it's a clever idea. However I am not sure this is testing the right thing.